File: hoiiday-backend/src/main/java/com/example/hoiiday/service/ClientService.java
package com.example.hoiiday.service;

import com.example.hoiiday.DTO.UserDTO;

import java.util.Set;

public interface ClientService {
    Set<UserDTO> getAllClients();
    UserDTO updateUser(Long userId, UserDTO userDTO);
    void deleteUser(Long userId);
}

File: hoiiday-front/hoiiday-frontend/src/app/services/client.service.ts
import { Injectable } from "@angular/core";
import { UserService } from "./user.service";
import { map, Observable } from "rxjs";
import { User } from "../model/user.model";

@Injectable({
    providedIn: 'root',
})
export class ClientService{
    constructor(private userService: UserService){}
    
    getClients(): Observable<User[]> {
    return this.userService.getAllUsers().pipe(
      map((users) => users.filter((user) => user.role === 'CLIENT'))
    );
    }

    deleteClient(userId: number): Observable<any>{
        return this.userService.deleteUser(userId);
    }

    // updateClient(user: User): Observable<User>{
    //     return this.userService.updateUser(user);
    // }
}

